packages/@aws-cdk/aws-appmesh/test/test.route.ts

```diff
@@ -282,6 +282,242 @@ export = {
+      });
+
+      // WHEN
+      test.throws(() => {
+        router.addRoute('test-http-route', {
+          routeSpec: appmesh.RouteSpec.http({
+            weightedTargets: [{ virtualNode }],
+            retryPolicy: {
+              maxRetries: 5,
+              perRetryTimeout: cdk.Duration.seconds(10),
+            },
+          }),
+        });
+      }, /specify one value for at least/i);
+
+      test.done();
+    },
+
+    'errors when retry event arrays are empty'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+      const virtualNode = mesh.addVirtualNode('test-node', {
+        serviceDiscovery: appmesh.ServiceDiscovery.dns('test'),
+        listeners: [appmesh.VirtualNodeListener.http()],
+      });
+
+      // WHEN
+      test.throws(() => {
+        appmesh.RouteSpec.http({
+          weightedTargets: [{ virtualNode }],
+          retryPolicy: {
+            httpRetryEvents: [],
+            tcpRetryEvents: [appmesh.TcpRetryEvent.CONNECTION_ERROR],
+            maxRetries: 5,
+            perRetryTimeout: cdk.Duration.seconds(10),
+          },
+        });
+      }, /at least one.*httpRetryEvents/);
+
+      test.throws(() => {
+        appmesh.RouteSpec.http({
+          weightedTargets: [{ virtualNode }],
+          retryPolicy: {
+            httpRetryEvents: [appmesh.HttpRetryEvent.CLIENT_ERROR],
+            tcpRetryEvents: [],
+            maxRetries: 5,
+            perRetryTimeout: cdk.Duration.seconds(10),
+          },
+        });
+      }, /at least one.*tcpRetryEvents/);
+
+      test.throws(() => {
+        appmesh.RouteSpec.grpc({
+          weightedTargets: [{ virtualNode }],
+          match: { serviceName: 'servicename' },
+          retryPolicy: {
+            grpcRetryEvents: [],
+            tcpRetryEvents: [appmesh.TcpRetryEvent.CONNECTION_ERROR],
+            maxRetries: 5,
+            perRetryTimeout: cdk.Duration.seconds(10),
+          },
+        });
+      }, /at least one.*grpcRetryEvents/i);
+
+      test.throws(() => {
+        appmesh.RouteSpec.grpc({
+          weightedTargets: [{ virtualNode }],
+          match: { serviceName: 'servicename' },
+          retryPolicy: {
+            httpRetryEvents: [],
+            tcpRetryEvents: [appmesh.TcpRetryEvent.CONNECTION_ERROR],
+            maxRetries: 5,
+            perRetryTimeout: cdk.Duration.seconds(10),
+          },
+        });
+      }, /at least one.*httpRetryEvents/i);
+
+      test.throws(() => {
+        appmesh.RouteSpec.grpc({
+          weightedTargets: [{ virtualNode }],
+          match: { serviceName: 'servicename' },
+          retryPolicy: {
+            httpRetryEvents: [appmesh.HttpRetryEvent.CLIENT_ERROR],
+            tcpRetryEvents: [],
+            maxRetries: 5,
+            perRetryTimeout: cdk.Duration.seconds(10),
+          },
+        });
+      }, /at least one.*tcpRetryEvents/i);
```

Comment:
Like I commented on the production code, I think this is not good behavior.

I would remove all of these tests in favor of checking that providing an empty list results in the appropriate value not being written to the template. You can use the `ABSENT` constant from the assertion module to assert that the given value is not used - see [example usage](https://github.com/aws/aws-cdk/blob/ba99ad086e309fe5d11c75351e4b5dbb1f199478/packages/%40aws-cdk/aws-apigatewayv2/test/http/api.test.ts#L130-L130).

Reply:
Cool. I didn't know this constant was a thing. Alright. This shouldn't be a problem.